server/athenian/api/controllers/reposet.py

```diff
@@ -50,7 +55,16 @@
     rs = await db.fetch_one(select(columns).where(RepositorySet.id == id))
     if rs is None or len(rs) == 0:
         raise ResponseError(NotFoundError(detail="Repository set %d does not exist" % id))
-    if rs[RepositorySet.owner.key] != user.id:
+    if account is None:
+        # if there is no account, suppose the reposet owner
+        account = rs[RepositorySet.owner.key]
+    elif rs[RepositorySet.owner.key] != account:
         raise ResponseError(ForbiddenError(
             detail="User %s is not allowed to access %d" % (user.id, id)))
-    return RepositorySet(**rs)
+    # finally check that the user belongs to the account
+    status = await db.fetch_one(select([UserAccount.is_admin]).where(
+        and_(UserAccount.user_id == user.id, UserAccount.account_id == account)))
+    if status is None:
+        raise ResponseError(ForbiddenError(
+            detail="User %s does not belong to account %d" % (user.id, account)))
```

Comment:
I'd split this logic into a separate `is_admin` function since it's also used in other parts.  For example, when modifying a reposet to me makes more sense to check first whether the client is an admin.

Reply:
Cool!